### PR TITLE
HYDRO-2200: /collections/{collection_id}/children/{child_id} GET

### PR DESCRIPTION
**Pull request by allalee**

### CODE REVIEW

File: components/parameters.yaml [hunk 1]
```diff
@@ -68,6 +68,18 @@ collection_id_query:
...
+  name: child_id
+  in: path
+  required: true
+  description: 'A child collection/gallery ID'
+  schema:
+    type: string
+    minLength: 16
+    maxLength: 16
+    pattern: '^[C|G]0000[a-zA-Z0-9._]{11}$'
+    example: C0000c1x4UMGPvkY
```
Comment (photoshelter-dev) [inline components/parameters.yaml]: I see you defined the child id in the parameters file, could you create a schema instead so we can reuse it?

File: components/parameters.yaml [hunk 2]
```diff
@@ -157,6 +169,22 @@ sort_direction:
...
+  required: false
+  style: form
+  explode: true
+  schema:
+    type: string
+    enum:
+    - asc
+    - desc
+  example: asc
+
```
Comment (photoshelter-dev) [inline components/parameters.yaml]: This definition already exists in this file under _sort_direction_

File: components/schemas.yaml [hunk 3]
```diff
@@ -72,6 +72,70 @@ Collection:
...
+          example: '2019-04-25T14:54:12.136828'
+          nullable: true
+    links:
+      type: object
+      properties:
+        self:
+          type: string
+          description: 'Link to itself'
+          example: 'https://api.photoshelter.com/psapi/v4/collections/C0000EXAMPLECOLL'
+
```
Comment (photoshelter-dev) [inline components/schemas.yaml]: How _ChildCollection_ is different than _Collection_ defined in this file?

File: components/schemas.yaml [hunk 4]
```diff
@@ -223,6 +287,68 @@ Gallery:
...
+          type: string
+          description: 'Last modified date'
+          example: '2019-04-25T14:54:12.136828'
+    links:
+      type: object
+      properties:
+        self:
+          type: string
+          description: 'Link to itself'
+          example: 'https://api.photoshelter.com/psapi/v4/galleries/G0000EXAMPLEGALL'
```
Comment (photoshelter-dev) [inline components/schemas.yaml]: Same as above, how is _ChildGallery_ different than _Gallery_ defined in this file?

File: components/schemas.yaml [hunk 5]
```diff
@@ -223,6 +287,68 @@ Gallery:
...
+        - description
+        - display_order
+        - is_listed
+        - sort_order
+        - created_at
+        - modified_at
+      properties:
+        name:
+          type: string
+          description: 'Name of Gallery or Collection object (Whichever it is)'
```
Comment (photoshelter-dev) [inline components/schemas.yaml]: Please update description to highlight galleries only. See jira ticket for more details.

File: paths/collectionsIdChildrenChildId.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  operationId: collectionsGetIdChildId
+  tags:
+    - Collection
+  description: >-
+    "Get information on a specific child of a collection. This endpoint does not require authentication. 
+    The ability to retrieve a specific child of a collection is dependent on the visibility and access 
+    options set to the parent collection. If a user does not have access to the collection then they 
+    will not be able to view a specific child of a collection. Some fields will be returned as null 
+    if the user does not have ownership access. The fields that are impacted are 'created_at', 
+    'access_id', modified_at'. If a child is not found with the id then a 404 error will be thrown."
```
Comment (photoshelter-dev) [inline paths/collectionsIdChildrenChildId.yaml]: When using multiline description, it is not necessary to use the double quotes.

File: components/schemas.yaml (hunk 3 above)
Comment (allalee) [reply components/schemas.yaml]: The ChildCollection and likewise the ChildGallery has a required "sort_order" attribute in the payload that would normally not exist.

File: components/parameters.yaml (hunk 1 above)
Comment (allalee) [reply components/parameters.yaml]: Will do.

File: components/parameters.yaml (hunk 2 above)
Comment (allalee) [reply components/parameters.yaml]: Not sure what I was thinking really...

File: paths/collectionsIdChildrenChildId.yaml (hunk 6 above)
Comment (allalee) [reply paths/collectionsIdChildrenChildId.yaml]: Fixed

File: components/schemas.yaml (hunk 3 above)
Comment (photoshelter-dev) [reply components/schemas.yaml]: I understand but you can use inheritance and model composition in this case.

See first example : https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/

**Pull request merged**